Main.hs

```diff
@@ -273,30 +276,66 @@ compileConstructorArgs (ExtendTel a tel)
     (:) <$> compileType (unEl $ unDom a) <*> compileConstructorArgs tel
 compileConstructorArgs tel = genericDocError =<< text "Bad constructor args:" <?> prettyTCM tel
 
-compileFun :: Definition -> TCM [Hs.Decl ()]
-compileFun def = do
+compilePostulate :: Definition -> TCM [Hs.Decl ()]
+compilePostulate def = do
   let x = hsName $ prettyShow $ qnameName $ defName def
   ty <- compileType (unEl $ defType def)
-  cs <- mapM (compileClause x) funClauses
+  let body = hsError $ "postulate: " ++ pp ty
+  return [ Hs.TypeSig () [x] ty
+         , Hs.FunBind () [Hs.Match () x [] (Hs.UnGuardedRhs () body) Nothing] ]
+  where
+    Axiom = theDef def
+
+type LocalDecls = [(QName, Definition)]
+
+groupDecls :: LocalDecls -> [(Definition, LocalDecls)]
+groupDecls [] = []
+groupDecls ((q, d) : defs) =
+  let getLvl (QName (MName xs) _) = length xs
+      (children, defs') = span ((> getLvl q) . getLvl . fst) defs
+  in (d, children) : groupDecls defs'
+
+compileFun :: Definition -> TCM [Hs.Decl ()]
+compileFun d = do
+  let findLocals = fromMaybe [] . fmap snd . find ((defName d ==) . defName . fst)
+  locals <- findLocals . groupDecls . sortDefs <$> curDefs
+  compileFun' locals d
```

Comment:
This does not work for functions with multiple clauses. Or rather, it might, but you need to do a reachability analysis in each clause to make sure you only get the local functions for each individual clause.